File: c++/leetcode/1346. Check If N and Its Double Exist.cpp
class Solution
{
public:
  bool checkIfExist(vector<int> &arr)
  {
    unordered_map<int, int> seen;

    for (auto i : arr)
      ++seen[i];

    for (auto i : seen)
    {
      int num = i.first;
      if (seen.find(num * 2) != seen.end())
      {
        if (num * 2 == num && seen[num] < 2)
          continue;
        return true;
      }
    }

    return false;
  }
};

File: javascript/leetcode/1346. Check If N and Its Double Exist.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var checkIfExist = function (arr) {
  const seen = new Map();

  arr.forEach((el) => {
    if (seen.get(el) === undefined || seen.get(el) === null) seen.set(el, 1);
    else seen.set(el, seen.get(el) + 1);
  });

  let ans = false;

  seen.forEach((val, key) => {
    if (seen.has(key * 2)) {
      if (key * 2 === key && val < 2) {
      } else ans = true;
    }
  });

  console.log(ans);

  return ans || false;
};
